core/src/main/java/org/apache/calcite/rel/rules/JoinToCorrelateRule.java

```diff
@@ -56,31 +60,57 @@
  *
  * <p>would require emitting a NULL emp row if a certain department contained no
  * employees, and Correlator cannot do that.</p>
+ *
+ * <p>There is a constructor parameterized to allow any sub-class of
```

Comment:
Since the parameterized constructor is going to be private we should not put this part of documentation here.